### PR TITLE
[Overflow-361] [FE] Implement API to User Detail Page for Admin

### PR DESCRIPTION
**Pull request by jules-lucero**
## Backlog Link
https://framgiaph.backlog.com/view/SUN_OVERFLOW-361

## Commands
Go to Admin User Detail Page or `http://localhost:3000/manage/users/<user-slug>`

## Pre-conditions
- You must have access to the page (You can comment `AdminWrapper` temporarily)

## Expected Output
- [x] You should be able to see the questions posted by a user
- [x] You should be able to see the answers posted by a user
- [x] You should be able to see the teams of a user
- [x] You should be able to switch from `List View` to `Grid View`
- [x] You should be able to filter the questions

## Notes
- Dependent on branch `feature/refactor-profile-page`

## Screenshots
Questions Tab (List View)
image
Questions Tab (Grid View)
image
Answers Tab (List View)
image
Answers Tab (Grid View)
image
Teams Tab
image

### CODE REVIEW

**Review by nicoleamber — DISMISSED** [review]

File: frontend/components/atoms/Button/index.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const getButtonClasses = (usage: string, size: string): string => {
         case 'icon':
             return 'absolute inset-y-0 right-0 mr-2 flex items-center'
         case 'popover':
-            return 'items-center rounded-lg border-2 px-5 py-2.5 text-center text-sm font-medium focus:ring-1 text-red-700 bg-white border-red-500 focus:ring-red-600 hover:bg-light-red'
+            return 'rounded-[5px] h-[28px] border-2 px-4 text-[12px] text-neutral-900 bg-white border-neutral-900 border-[1px] hover:bg-neutral-200 outline-none'
```
Comment (IanUrriza10) [inline frontend/components/atoms/Button/index.tsx:62]:
```suggestion
            return 'rounded-[5px] h-7 border-2 px-4 text-xs text-neutral-900 bg-white border-neutral-900 border-[1px] hover:bg-neutral-200 outline-none'
```

File: frontend/components/molecules/Tag/index.tsx [hunk 2]
```diff
@@ -9,16 +11,29 @@ type TagProps = {
...
-                    <EyeIcon />
+        <Popover
+            placement="bottom-start"
+            animate={{
+                mount: { scale: 1, y: 0 },
+                unmount: { scale: 0, y: 25 },
+            }}
+        >
+            <PopoverHandler>
+                <div className="flex h-4 cursor-pointer items-center gap-[2px] rounded-xl bg-neutral-200 px-1 py-[2px] text-[10px]">
```
Comment (IanUrriza10) [inline frontend/components/molecules/Tag/index.tsx:22]:
```suggestion
                <div className="flex h-4 cursor-pointer items-center gap-0.5 rounded-xl bg-neutral-200 px-1 py-0.5 text-[10px]">
```

File: frontend/components/molecules/TagPopover/index.tsx [hunk 3]
```diff
@@ -48,43 +73,32 @@ const Tooltips = ({ tag }: TagType): JSX.Element => {
...
-                    <Link
-                        href={`/questions/tagged/${tag.slug ?? ''}`}
-                        className="ml-2 text-blue-500 underline hover:text-blue-400"
-                    >
-                        View Tag
-                    </Link>
-                </span>
+        <div className="flex min-h-[88px] w-72 flex-col gap-2">
+            <div className="border-b border-neutral-300 px-2 pb-2">
+                <span className="text-[12px] text-neutral-900">{tag.description}</span>
```
Comment (IanUrriza10) [inline frontend/components/molecules/TagPopover/index.tsx:79]:
```suggestion
                <span className="text-xs text-neutral-900">{tag.description}</span>
```

File: frontend/components/organisms/QuestionGridItem/index.tsx [hunk 4]
```diff
@@ -1,95 +1,76 @@
...
-                        Lorem ipsum dolor sit amet consectetur. Sed amet at id sit proin in. Lorem
-                        ipsum dolor sit amet consectetur. Sed amet at id sit proin in. Lorem ipsum
-                        dolor sit amet consectetur. Sed amet at id sit proin in. Lorem ipsum dolor
-                        sit amet consectetur. Sed amet at id sit proin in. Lorem ipsum dolor sit
-                        amet consectetur. Sed amet at id sit proin in.
-                    </p>
-                    <div className="flex w-fit gap-1 rounded-[4px] border border-primary-base px-1 py-[2px] text-primary-base">
+            <div className="flex flex-1 gap-3">
+                <div className="flex flex-1 flex-col gap-2">
+                    <p className="text-[12px] leading-4 line-clamp-3">{parseHTML(content)}</p>
```
Comment (IanUrriza10) [inline frontend/components/organisms/QuestionGridItem/index.tsx]:
```suggestion
                    <p className="text-xs leading-4 line-clamp-3">{parseHTML(content)}</p>
```

**Review by IanUrriza10 — DISMISSED** [review]

**Review by IanUrriza10 — APPROVED** [review]

**Review by kenorenzsun — DISMISSED** [review]

**Review by IanUrriza10 — DISMISSED** [review]

**Comment by IanUrriza10** [comment]
## Approved PR
- All displayed the same as the Figma design
- Borders cannot be reduced from 1 to 0.5 for the bottom of tabs and border of view and filter
- 
## Details
- in `/manage/users/slug` there are 3 tabs for questions, answers, and teams
 - Questions - Lists all the questions sent by the user (newest first)
 - Answers - Lists all the answers sent by the user (newest first)
 - Teams - Lists all the teams the user belongs too (newest created team first)
 - Will display an isEmpty message if none were found
 - All Question/Answer Card will redirect to the proper question

## Notes
 - Creating Teams will instantly insert the Team Leader in Members Table with role_id 1
 - Query for questions is modified due to it only taking public questions
 
## Proof

https://user-images.githubusercontent.com/114897466/234796957-06979eb8-10be-49fd-9254-62f9ce3b0236.mp4

- Total Test time: 3 hr.
- Total Comments: 3
- Total Iterations: 3

**Review by IanUrriza10 — APPROVED** [review]

**Review by kenorenzsun — APPROVED** [review]

**Review by nicoleamber — APPROVED** [review]

**Pull request merged**